ts/IterableInbox.tsx

```diff
@@ -109,6 +111,42 @@ const IterableInbox = ({
       }
    }, [width])
 
+   if(appState === 'active') {
+      if(isFocused) {
+         inboxDataModel.startSession()
+         inboxDataModel.updateVisibleRows(visibleRowImpressions)
+      } else {
+         inboxDataModel.endSession()
+      }
+   }
+
+   function getInitialMessageImpressions(messageListHeight: number, messageRowHeight: number, rowViewModels: InboxRowViewModel[]) {
+      let numMessages = Math.floor(messageListHeight / messageRowHeight)
+      let visibleRowViewModels = rowViewModels.slice(0, numMessages)
+
+      setVisibleRowImpressions(visibleRowViewModels.map(rowViewModel => {
```

Comment:
The translation inside that `map` should definitely get separated into its own function.

Reply:
I realized I wrote `var inAppMessage = IterableInAppMessage.fromViewToken(viewToken)` exactly to replace the giant block of text. Is there a reason to not use that?

Reply:
Yes, sounds good.